test/uk/gov/hmrc/excisemovementcontrolsystemapi/scheduling/NrsSubmissionSchedulerSpec.scala

```diff
@@ -0,0 +1,164 @@
+import uk.gov.hmrc.excisemovementcontrolsystemapi.repository.model.NrsSubmissionWorkItem
+import uk.gov.hmrc.mongo.workitem.ProcessingStatus.ToDo
+import uk.gov.hmrc.mongo.workitem.{ProcessingStatus, WorkItem}
+
+import java.time.{Instant, ZonedDateTime}
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+import scala.concurrent.duration.FiniteDuration
+
+class NrsSubmissionSchedulerSpec
+    extends AnyFreeSpec
+    with Matchers
+    with ScalaFutures
+    with IntegrationPatience
+    with GuiceOneAppPerSuite
+    with BeforeAndAfterEach
+    with NrsTestData {
+
+  val mockNrsSubmissionWorkItemRepository: NRSWorkItemRepository = mock[NRSWorkItemRepository]
+  val mockNrsConnector: NrsConnector                             = mock[NrsConnector]
+  val appConfig: AppConfig                                       = app.injector.instanceOf[AppConfig]
+  private lazy val nrsSubmissionScheduler                        = app.injector.instanceOf[NrsSubmissionScheduler]
+  private val now                                                = Instant.now
+  private val timeStamp                                          = ZonedDateTime.now()
+  private val nrsMetadata                                        = NrsMetadata(
+    businessId = "emcs",
+    notableEvent = "excise-movement-control-system",
+    payloadContentType = "application/json",
+    payloadSha256Checksum = sha256Hash("payload for NRS"),
+    userSubmissionTimestamp = timeStamp.toString,
+    identityData = testNrsIdentityData,
+    userAuthToken = testAuthToken,
+    headerData = Map(),
+    searchKeys = Map("ern" -> "123")
+  )
+  private val nrsPayLoad                                         = NrsPayload("encodepayload", nrsMetadata)
+  override def fakeApplication(): Application                    = GuiceApplicationBuilder()
+    .overrides(
+      bind[NRSWorkItemRepository].toInstance(mockNrsSubmissionWorkItemRepository),
+      bind[NrsConnector].toInstance(mockNrsConnector)
+    )
+    .configure(
+      "scheduler.nrsSubmissionJob.initialDelay" -> "1 minutes",
+      "scheduler.nrsSubmissionJob.interval"     -> "1 minute",
+      "featureFlags.nrsSubmissionEnabled"       -> true
+    )
+    .build()
+
+  override def beforeEach(): Unit = {
+    super.beforeEach()
+    Mockito.reset[Any](
+      mockNrsSubmissionWorkItemRepository,
+      mockNrsConnector
+    )
+  }
+
+  "nrs submission scheduler must" - {
+    "have the correct name" in {
+      nrsSubmissionScheduler.name mustBe "nrs-submission-scheduler"
+    }
+    "be enabled" in {
+      nrsSubmissionScheduler.enabled mustBe true
+    }
+    "use initial delay from configuration" in {
+      nrsSubmissionScheduler.initialDelay mustBe FiniteDuration(1, "minutes")
+    }
+    "use interval from configuration" in {
+      nrsSubmissionScheduler.interval mustBe FiniteDuration(1, "minute")
+    }
+  }
+
+  ".execute" - {
```

Comment:
Looks good, my only concern is there isn't any test around retrying a failed work item.

We should have a scenario of 'this failed too many times, pagerduty' which can leverage failureCount

Reply:
That will be added in the next PR :) but yes, agreed.

Reply:
Right, I'll consider it resolved then